src/color-swatch/color-swatch.js

```diff
@@ -29,236 +28,119 @@ const Self = class ColorSwatch extends HTMLElement {
+		let input = this._el.input;
 
-				// Is contained within scope root
-				if (scopeRoot) {
-					this.style.setProperty("--color", varRef);
+		if (name === "gamuts") {
+			if (this.gamuts === "none") {
+				this._el.gamutIndicator?.remove();
+				this._el.gamutIndicator = null;
+			}
+			else if (this.gamuts) {
+				if (!this._el.gamutIndicator) {
+					this._el.colorWrapper.insertAdjacentHTML("beforeend", `
+						<gamut-badge id="gamut" part="gamut" exportparts="label: gamutlabel" gamuts="${ this.gamuts }"></gamut-badge>
+					`);
+
+					this._el.gamutIndicator = this._el.colorWrapper.lastElementChild;
+
+					this._el.gamutIndicator.addEventListener("gamutchange", evt => {
+						let gamut = this._el.gamutIndicator.gamut;
+						this.setAttribute("gamut", gamut);
+						this.dispatchEvent(new CustomEvent("gamutchange", {
+							detail: gamut,
+						}));
+					});
 				}
-
-				scopeRoot ??= document.querySelector(this.scope);
-
-				if (scopeRoot) {
-					let cs = getComputedStyle(scopeRoot);
-					this.value = cs.getPropertyValue(this.property);
+				else {
+					this._el.gamutIndicator.gamuts = this.gamuts;
 				}
 			}
 		}
-	}
-
-	#render (evt) {
-		if (!this.#initialized) {
-			return;
-		}
-
-		clearTimeout(this.#errorTimeout);
-
-		if (!this.isConnected) {
-			return;
-		}
-
-		let value = this.value;
-		this.#color = null;
-
-		if (value) {
-			try {
-				this.#color = new Color(value);
-			}
-			catch (e) {
-				// Why a timeout? We don't want to produce errors for intermediate states while typing,
-				// but if this is a genuine error, we do want to communicate it.
-				this.#errorTimeout = setTimeout(_ => this.#dom.input?.setCustomValidity(e.message), 500);
-			}
 
-			if (this.#color) {
-				this.#setColor(this.#color);
-				this.#dom.input?.setCustomValidity("");
+		if (name === "value") {
+			if (!input.value) {
+				input.value = this.value;
 			}
-
-			this.dispatchEvent(new CustomEvent("colorchange", {
-				detail: {
-					color: this.#color,
-				},
-			}));
 		}
-	}
-
-	get gamut () {
-		return this.#dom.gamutIndicator.gamut;
-	}
 
-	get value () {
-		return this.#dom.input?.value ?? this.textContent.trim();
-	}
+		if (name === "color") {
+			let isValid = this.color !== null || !this.value;
 
-	set value (value) {
-		let oldValue = this.value;
-		if (value === oldValue) {
-			return;
-		}
+			input?.setCustomValidity(isValid ? "" : "Invalid color");
 
-		this.#setValue(value);
-		this.#render();
-	}
-
-	#setValue (value) {
-		if (!this.#initialized) {
-			this.#initialize();
-		}
-
-		if (this.#dom.input) {
-			this.#dom.input.value = value;
-		}
-		else {
-			this.textContent = value;
-		}
-	}
-
-	#color;
-	get color () {
-		return this.#color;
-	}
+			if (this._el.gamutIndicator) {
+				this._el.gamutIndicator.color = this.color;
+			}
 
-	set color (color) {
-		if (typeof color === "string") {
-			color = new Color(color);
+			let colorString = this.color?.display();
```

Comment:
For some reason, `color` is a string when we come here (probably because its default prop is `value`), so we don't have (on type `string`) the `display()` method we are trying to invoke. So, this line throws.

I wonder whether `this.color` should be of type `Color` here, right? But it's not. Hm. Need to investigate further.

Reply:
Oops. I didn't update `NudeElement`. Works flawlessly!